compare_vcf.py

```diff
@@ -369,32 +369,25 @@ def match_false(augmented_file, files_to_pair_with, out_dir, sample, log_to_file
                             shutil.rmtree(vcfeval_prefix)
 
                     if i == 0:
-                        AO_RO_DP = {"AO": None, "RO": None, "DP": None}
+                        AO_RO_DP_AD = {"AO": None, "RO": None, "DP": None, "AD": None}
                         if equivalent_variant:
-                            for entry in AO_RO_DP:
-                                try:
-                                    #Loop backwards through fields to preferentially extract AO and RO from SAMPLE over INFO
-                                    for field in equivalent_variant[::-1]:
-                                        found = re.search("['\t', ':']{}['\t', ':']".format(entry), field)
-                                        if found:
-                                            field_split = field.split(':')
-                                            entry_index = field_split.index('{}'.format(entry))
-                                            field_index = equivalent_variant.index(field)
-                                            AO_RO_DP[entry] = int(equivalent_variant[field_index+1].split(':')[entry_index])
-                                            break
-
-                                        found = re.search("['\t', ';']{}=".format(entry), field)
-                                        if found:
-                                            AO_RO_DP[entry] = int(re.split("['\t', ';']{}=".format(entry), field)[1].split(';')[0])
-                                            break
-                                except:
-                                    pass
-                        if AO_RO_DP["AO"] and AO_RO_DP["RO"]:
-                            info = str(float(AO_RO_DP["AO"])/(AO_RO_DP["AO"]+AO_RO_DP["RO"])) + ';'
+                            for entry in AO_RO_DP_AD:
+                                AO_RO_DP_AD[entry] = utils.get_info(equivalent_variant, entry)
+
+                        # gatk4 format
+                        if AO_RO_DP_AD["AD"]:
+                            AD_split = AO_RO_DP_AD["AD"].split(',')
+                            AO = int(AD_split[1])
```

Comment:
There could be more than 1 AO number and multiple ALT alleles, we need to take care of that.

Reply:
Done